File: frontend/src/types/datasource.ts
export interface TableInfo {
  name: string;
}

export interface ColumnInfo {
  name: string;
  dtype: string;
}

export interface SchemaOut {
  table_name: string;
  columns: ColumnInfo[];
}

export interface SampleRowsOut {
  table_name: string;
  rows: Record<string, unknown>[];
}

File: app/schemas/datasource.py
from typing import Any

from pydantic import BaseModel


class TableInfo(BaseModel):
    name: str


class ColumnInfo(BaseModel):
    name: str
    dtype: str


class SchemaOut(BaseModel):
    table_name: str
    columns: list[ColumnInfo]


class SampleRowsOut(BaseModel):
    table_name: str
    rows: list[dict[str, Any]]
